provider.go

```diff
@@ -342,10 +436,10 @@ func (p *Provider) MountSecretsStoreObjectContent(ctx context.Context, attrib ma
 			return err
 		}
 		objectContent := []byte(content)
-		if err := ioutil.WriteFile(path.Join(targetPath, keyValueObject.ObjectPath), objectContent, permission); err != nil {
-			return errors.Wrapf(err, "secrets-store csi driver failed to write %s at %s", keyValueObject.ObjectPath, targetPath)
+		if err := ioutil.WriteFile(path.Join(targetPath, keyValueObject.ObjectName), objectContent, permission); err != nil {
+			return errors.Wrapf(err, "secrets-store csi driver failed to write %s at %s", keyValueObject.ObjectName, targetPath)
```

Comment:
Changing from `ObjectPath` to `ObjectName` changes the existing behavior of this provider. Currently when user deploys this:

```yaml
apiVersion: secrets-store.csi.k8s.com/v1alpha1
kind: SecretProviderClass
metadata:
  name: vault-foo
spec:
  provider: vault
  parameters:
    roleName: "example-role"
    vaultAddress: http://${VAULT_SERVICE_IP}:8200
    vaultSkipTLSVerify: "true"
    objects:  |
      array:
        - |
          objectPath: "/foo"
          objectName: "bar"
          objectVersion: ""
```
From the container that uses the existing Vault provider, it will expect the value of `bar` to be written into `mountPath: "/mnt/secrets-store"` at `/mnt/secrets-store/foo`.

This change will change this behavior into the value of `bar` is written in `/mount/secrets-store/bar`.

Reply:
I think this is a reasonable change. It not only avoids the issue of trying to write nested keys to a directory that may not exist, but also lets someone access multiple values within a single key (e.g both "bar" and "baz" within the "/foo" key). Currently the last object in the array would win.

If the path aspect is desirable, we could look at a separate enhancement to prefix the output filename in a manner similar to [envconsul](https://github.com/hashicorp/envconsul#vault). This would result in a filename like "foo_bar" or "foo_subkey_bar". This is could be an opt-in enhancement.

Reply:
SGTM. 
@carnei-ro can you pls add some notes about this change in the readme as this is a breaking change. 

@kalafut @aramase after a new release of the vault provider is cut and updated in the driver, we need to update the e2e test in the driver repo to reflect this change as the e2e test as of now will fail.